src/execution_plan/ops/op_delete.c

```diff
@@ -139,38 +138,36 @@ OpBase *NewDeleteOp(const ExecutionPlan *plan, AR_ExpNode **exps) {
 	op->exp_count = array_len(exps);
 	op->deleted_nodes = array_new(Node, 32);
 	op->deleted_edges = array_new(Edge, 32);
+	op->records = NULL;
 
-	// Set our Op operations
+	// set our Op operations
 	OpBase_Init((OpBase *)op, OPType_DELETE, "Delete", NULL, DeleteConsume,
-				DeleteReset, NULL, DeleteClone, DeleteFree, true, plan);
+				NULL, NULL, DeleteClone, DeleteFree, true, plan);
 
 	return (OpBase *)op;
 }
 
-static Record DeleteConsume(OpBase *opBase) {
+// collect nodes and edges to be deleted
+static inline void _CollectDeletedEntities(Record r, OpBase *opBase) {
 	OpDelete *op = (OpDelete *)opBase;
-	OpBase *child = op->op.children[0];
 
-	Record r = OpBase_Consume(child);
-	if(!r) return NULL;
-
-	// Expression should be evaluated to either a node, an edge or a path
+	// expression should be evaluated to either a node, an edge or a path
 	// which will be marked for deletion, if an expression is evaluated
-	// to a different value type e.g. Numeric a run-time exception is thrown. */
+	// to a different value type e.g. Numeric a run-time exception is thrown.
 	for(int i = 0; i < op->exp_count; i++) {
 		AR_ExpNode *exp = op->exps[i];
 		SIValue value = AR_EXP_Evaluate(exp, r);
 		SIType type = SI_TYPE(value);
-		/* Enqueue entities for deletion. */
+		// enqueue entities for deletion
 		if(type & T_NODE) {
 			Node *n = (Node *)value.ptrval;
 			array_append(op->deleted_nodes, *n);
-			// If evaluating the expression allocated any memory, free it.
+			// if evaluating the expression allocated any memory, free it.
 			SIValue_Free(value);
 		} else if(type & T_EDGE) {
 			Edge *e = (Edge *)value.ptrval;
 			array_append(op->deleted_edges, *e);
-			// If evaluating the expression allocated any memory, free it.
+			// if evaluating the expression allocated any memory, free it.
 			SIValue_Free(value);
```

Comment:
Although a good practice, I think this is redundant, `Edge` typed SIValues aren't "self" allocated.